httpd/handler.go

```diff
@@ -183,6 +186,126 @@ func (h *Handler) serveQuery(w http.ResponseWriter, r *http.Request, user *influ
+}
+
+// Return all the measurements from the given DB
+func (h *Handler) showMeasurements(db string, user *influxdb.User) ([]string, error) {
+	var measurements []string
+	results := h.server.ExecuteQuery(&influxql.Query{[]influxql.Statement{&influxql.ShowMeasurementsStatement{}}}, db, user)
+	if results.Err != nil {
+		return measurements, results.Err
+	}
+
+	for _, result := range results.Results {
+		for _, row := range result.Series {
+			for _, tuple := range (*row).Values {
+				for _, cell := range tuple {
+					measurements = append(measurements, interfaceToString(cell))
+				}
+			}
+		}
+	}
+	return measurements, nil
+}
+
+// serveDump returns all points in the given database as a plaintext list of JSON structs.
+// To get all points:
+// Find all measurements (show measurements).
+// For each measurement do select * from <measurement> group by *
+func (h *Handler) serveDump(w http.ResponseWriter, r *http.Request, user *influxdb.User) {
+	q := r.URL.Query()
+	db := q.Get("db")
+	pretty := q.Get("pretty") == "true"
+	delim := []byte("\n")
+	measurements, err := h.showMeasurements(db, user)
+	if err != nil {
+		httpError(w, "error with dump: "+err.Error(), pretty, http.StatusInternalServerError)
+		return
+	}
+
+	// Fetch all the points for each measurement.
+	// From the 'select' query below, we get:
+	//
+	// columns:[col1, col2, col3, ...]
+	// - and -
+	// values:[[val1, val2, val3, ...], [val1, val2, val3, ...], [val1, val2, val3, ...]...]
+	//
+	// We need to turn that into multiple rows like so...
+	// fields:{col1 : values[0][0], col2 : values[0][1], col3 : values[0][2]}
+	// fields:{col1 : values[1][0], col2 : values[1][1], col3 : values[1][2]}
+	// fields:{col1 : values[2][0], col2 : values[2][1], col3 : values[2][2]}
+	//
+	for _, measurement := range measurements {
+		queryString := fmt.Sprintf("select * from %s group by *", measurement)
+		p := influxql.NewParser(strings.NewReader(queryString))
+		query, err := p.ParseQuery()
+		if err != nil {
+			httpError(w, "error with dump: "+err.Error(), pretty, http.StatusInternalServerError)
+			return
+		}
+		//
+		results := h.server.ExecuteQuery(query, db, user)
+		for _, result := range results.Results {
+			for _, row := range result.Series {
+				points := make([]Point, 1)
+				var point Point
+				point.Name = row.Name
+				point.Tags = row.Tags
+				point.Fields = make(map[string]interface{})
+				for _, tuple := range row.Values {
+					for subscript, cell := range tuple {
+						if row.Columns[subscript] == "time" {
+							point.Timestamp, _ = cell.(time.Time)
+							continue
+						}
+						point.Fields[row.Columns[subscript]] = cell
+					}
+					points[0] = point
+					batch := &Batch{
+						Points:          points,
+						Database:        db,
+						RetentionPolicy: "default",
+					}
+					buf, err := json.Marshal(&batch)
+
+					if err != nil {
+						httpError(w, "error with dump: "+err.Error(), pretty, http.StatusInternalServerError)
```

Comment:
I only added this at your insistence that this endpoint handle the .Marshal errors even though the other endpoints don't. Please see your previous comment on this section of code. 

Since the chunked respsonse is well underway at this point, I don't see a good way to signal to the client that something has gone wrong.

Please just decide what you want.